taropsbt/interface.go

```diff
@@ -0,0 +1,141 @@
+package taropsbt
+
+import (
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taro/asset"
+	"github.com/lightninglabs/taro/commitment"
+)
+
+// VPacket is a PSBT extension packet for a virtual transaction. It represents
+// the virtual asset state transition as it will be validated by the Taro VM.
+// Some elements within the virtual packet may refer to on-chain elements (such
+// as the anchor BTC transaction that was used to anchor the input that is
+// spent). But in general a virtual transaction does NOT directly map onto a BTC
+// transaction. It is entirely possible that multiple virtual transactions will
+// be merged into a single BTC transaction. Thus, each asset state transfer is
+// represented in a virtual TX and multiple asset state transfers can be
+// anchored within a single BTC transaction.
+//
+// TODO(guggero): Actually support merging multiple virtual transactions into a
+// single BTC transaction.
+type VPacket struct {
+	// Input is the asset input that is being spent.
+	//
+	// TODO(guggero): Support spending multiple inputs.
+	Input *VInput
+
+	// Outputs is the list of new asset outputs that are created by the
+	// virtual transaction. By convention the output at index 0 is the
+	// change output for asset change or the asset tombstone in case of a
+	// non-interactive full value send.
+	Outputs []*VOutput
+}
+
+// VInput represents an input to a virtual asset state transition transaction.
+type VInput struct {
+	// PInput is the embedded default PSBT input struct that is used for
+	// already existing data such as the Taproot internal key or script
+	// witness.
+	psbt.PInput
+
+	// PrevID is the asset previous ID of the asset being spent.
+	PrevID asset.PrevID
+
+	// AnchorPoint is the outpoint that the Commitment below is anchored on
+	// in the main chain. When serialized, this will be stored as the
+	// PreviousOutPoint of the wire.TxIn of the PSBT's unsigned TX.
+	AnchorPoint wire.OutPoint
+
+	// AnchorOutputValue is output value of the anchor output.
+	AnchorOutputValue btcutil.Amount
+
+	// AnchorOutputInternalKey is the internal key of the anchor output that
+	// this input is spending the asset from.
+	AnchorOutputInternalKey *btcec.PublicKey
+
+	// AnchorOutputBip32Derivation is the BIP32 derivation of the anchor
+	// output's internal key.
+	//
+	// TODO(guggero): Do we also want to allow multiple derivations to be
+	// specified here? That would allow us to specify multiple keys involved
+	// in MuSig2 for example. Same for the Taproot derivation below.
+	AnchorOutputBip32Derivation *psbt.Bip32Derivation
+
+	// AnchorOutputTaprootBip32Derivation is the Taproot BIP32 derivation of
+	// the anchor output's internal key.
+	AnchorOutputTaprootBip32Derivation *psbt.TaprootBip32Derivation
```

Comment:
The thing is, since we might need to sign for this key, we need the full, 33-byte internal key so we can use it for the `lnd` APIs. And the Taproot version only contains the 32-byte x-only key, which means we'd not be able to derive the internal key in 50% of the times. Unless we only rely on the derivation information? But I think even there the `lnd` PSBT APIs also want the non-Taproot version to be set.